Source/WebKit/WebProcess/WebPage/Cocoa/WebPageCocoa.mm

```diff
@@ -311,14 +311,21 @@
 }
 
 #if ENABLE(UNIFIED_TEXT_REPLACEMENT)
-void WebPage::getTextIndicatorForID(const WTF::UUID& uuid, CompletionHandler<void(std::optional<WebCore::TextIndicatorData>&&)>&& completionHandler)
+std::optional<WebCore::SimpleRange> WebPage::getRangeforUUID(const WTF::UUID& uuid)
 {
-    auto sessionRange = m_unifiedTextReplacementController->contextRangeForSessionWithUUID(uuid);
-    if (!sessionRange) {
+    auto range = m_unifiedTextReplacementController->contextRangeForSessionWithUUID(uuid);
```

Comment:
this can probably be inverted to have less nested branches, and I think `makeSimpleRange` null checks the parameter too.

```
auto range = m_unifiedTextReplacementController->contextRangeForSessionWithUUID(uuid);
if (range)
    return range;

RefPtr liveRange = m_textIndicatorStyleEnablementRanges.get(uuid);

return WebCore::makeSimpleRange(liveRange);